server/internal/core/application/utils.go

```diff
@@ -32,14 +32,16 @@ type timedTxRequest struct {
 }
 
 type txRequestsQueue struct {
-	lock     *sync.RWMutex
-	requests map[string]*timedTxRequest
+	lock      *sync.RWMutex
+	requests  map[string]*timedTxRequest
+	vtxoIndex map[domain.VtxoKey]struct{}
```

Comment:
can we move this directly to the app svc and make it become of type redeemTxRequests (that's why i suggested the renaming)? This way we not make this double-check operation dependant from the txRequestQueue that is meant for handling in-round swaps